e2e_tests/suite/test_smoke.py

```diff
@@ -0,0 +1,114 @@
+            print(f"Scaling activity started: {response}")
+            return response
+        except autoscaling_client.exceptions.ScalingActivityInProgressFault:
+            print("Scaling activity is in progress, wait for 60 seconds then retry.")
+            counter = counter + 1
+            time.sleep(RECONFIGURATION_DELAY)
+    pytest.fail(f"Failed to scale the group {group_name}")
+
+
+class TestSmoke:
+    @pytest.mark.parametrize("test_data", [
+        pytest.param({'group_name': 'WebserverGroup1', 'api_url': '/http/upstreams/backend1'}, id="backend1"),
+        pytest.param({'group_name': 'WebserverGroup2', 'api_url': '/http/upstreams/backend2'}, id="backend2"),
+        pytest.param({'group_name': 'WebserverGroup3', 'api_url': '/stream/upstreams/tcp-backend'}, id="tcp-backend")
+    ])
+    def test_aws_scale_up(self, cli_arguments, autoscaling_client, test_data):
+        desired_capacity = 5
+        groups = autoscaling_client.describe_auto_scaling_groups()['AutoScalingGroups']
+        group_name = list(filter(
+                lambda group: test_data['group_name'] in group['AutoScalingGroupName'], groups
+            ))[0]['AutoScalingGroupName']
+        scale_aws_group(autoscaling_client, group_name, desired_capacity)
+        wait_for_changes_in_aws(autoscaling_client, group_name, desired_capacity)
+        wait_for_changes_in_api(f"{cli_arguments.nginx_api}/{NGINX_API_VERSION}{test_data['api_url']}",
+                                desired_capacity)
+        resp = requests.get(f"{cli_arguments.nginx_api}/{NGINX_API_VERSION}{test_data['api_url']}")
+        nginx_upstream = json.loads(resp.text)
+        assert len(nginx_upstream['peers']) == desired_capacity,\
+            f"Expected {desired_capacity} servers, found: {nginx_upstream['peers']}"
+
+    @pytest.mark.parametrize("test_data", [
+        pytest.param({'group_name': 'WebserverGroup1', 'api_url': '/http/upstreams/backend1'}, id="backend1"),
+        pytest.param({'group_name': 'WebserverGroup2', 'api_url': '/http/upstreams/backend2'}, id="backend2"),
+        pytest.param({'group_name': 'WebserverGroup3', 'api_url': '/stream/upstreams/tcp-backend'}, id="tcp-backend")
+    ])
+    def test_aws_scale_down(self, cli_arguments, autoscaling_client, test_data):
+        desired_capacity = 1
+
+        groups = autoscaling_client.describe_auto_scaling_groups()['AutoScalingGroups']
+        group_name = list(filter(
+                lambda group: test_data['group_name'] in group['AutoScalingGroupName'], groups
+            ))[0]['AutoScalingGroupName']
+        scale_aws_group(autoscaling_client, group_name, desired_capacity)
+        wait_for_changes_in_aws(autoscaling_client, group_name, desired_capacity)
+        wait_for_changes_in_api(f"{cli_arguments.nginx_api}/{NGINX_API_VERSION}{test_data['api_url']}",
+                                desired_capacity)
+        resp = requests.get(f"{cli_arguments.nginx_api}/{NGINX_API_VERSION}{test_data['api_url']}")
```

Comment:
this code is already  present in `wait_for_changes_in_api`. Why do we need to call it again? Can `wait_for_changes_in_api` just fail after N number of attempts?  same applies to the other function.

Reply:
I prefer to ensure that AWS part is done successfully before make an assertion for ASG tool. Otherwise it's hard to understand where the problem is.